### PR TITLE
Encrypted profile pic

### PR DESCRIPTION
**Pull request by sachaaaaa**
- Encrypt/decrypt picture when uploading/downloading
- Prevent downloading our own picture right after having uploaded it (we already have the bytes)
- ~~set `profileSharing` on every conversation to true to enable sending the `profileKey` with every message~~

~~Note that the public chat side of things will come in the next PR.~~
~~EDIT: changes for public chat added in this PR but in a separate commit~~

- set `profileSharing`  to true only before sending a friend request (FR), or after becoming friend, for private chats
- add logic in the AppDotNet API to store our profileKey in the user annotation
- ~~share our profileKey on the public chat only upon sending a message the first time~~
- ~~share our profileKey on the public chat every time it changes, unless we've never sent any messages~~

EDIT2: Major re-org of this PR, please review commit by commit :)

### CODE REVIEW

**Comment by sachaaaaa** [comment]
I doubt this PR tackles the multi-device case, where we show the primary device's avatar for a message sent by a secondary device. Am testing this now, will post any fixes in a new PR if necessary!

**Comment by sachaaaaa** [comment]
I have force-pushed a brand new history for this branch.
This should have everything for encrypted profile pics, with all specific changes split into separate commits.

File: js/modules/loki_app_dot_net_api.js [hunk 1]
```diff
@@ -254,6 +266,14 @@ class LokiAppDotNetServerAPI {
     return res.response.data.annotations || [];
   }
 
+  async setAvatar(url, profileKey) {
+    let value = null;
+    if (url && profileKey) {
```
Comment (Mikunj) [inline js/modules/loki_app_dot_net_api.js]: would it be good to check for an empty url here too?

File: js/modules/loki_app_dot_net_api.js [hunk 2]
```diff
@@ -1294,9 +1322,12 @@ class LokiPublicChannelAPI {
         if (slavePrimaryMap[messageData.source]) {
           // rewrite source, profile
           messageData.source = primaryPubKey;
-          messageData.message.profile.displayName = this.primaryUserProfileName[
+          const { name, avatar, profileKey } = this.primaryUserProfileName[
             primaryPubKey
           ];
+          messageData.message.profile.displayName = name;
```
Comment (Mikunj) [inline js/modules/loki_app_dot_net_api.js]: `messageData.message` can just become an object.
```
messageData.message = {
    profile: { ... }
    profileKey: ....
}

File: js/modules/loki_app_dot_net_api.js [hunk 3]
```diff
@@ -254,6 +266,14 @@ class LokiAppDotNetServerAPI {
     return res.response.data.annotations || [];
   }
 
+  async setAvatar(url, profileKey) {
+    let value = null;
```
Comment (neuroscr) [inline js/modules/loki_app_dot_net_api.js]: This needs to be undefined. A null is preserved but an undefined will clean up the complete annotation. This is post Sacha, I can change this after the PR.

File: js/modules/loki_file_server_api.js [hunk 4]
```diff
@@ -17,6 +17,10 @@ class LokiFileServerInstance {
     this._adnApi = new LokiAppDotNetAPI(ourKey);
     this.avatarMap = {};
   }
+
+  // FIXME: this is not file-server specific
+  // and is currently called by LokiAppDotNetAPI.
+  // LokiAppDotNetAPI (base) should not know about LokiFileServer.
```
Comment (neuroscr) [inline js/modules/loki_file_server_api.js]: This needs to be discussed

**Pull request merged**